examples/inference-deployments/mpt/mpt_7b_ft_handler.py

```diff
@@ -0,0 +1,248 @@
+                generate_kwargs[k] = v
+        generate_kwargs['top_k'] *= torch.ones(batch_size, dtype=torch.int32)
+        generate_kwargs['top_p'] *= torch.ones(batch_size, dtype=torch.float32)
+        generate_kwargs['temperature'] *= torch.ones(batch_size,
+                                                     dtype=torch.float32)
+        repetition_penalty = generate_kwargs['repetition_penalty']
+        generate_kwargs[
+            'repetition_penalty'] = None if repetition_penalty == 1.0 else repetition_penalty * torch.ones(
+                batch_size, dtype=torch.float32)
+        presence_penalty = generate_kwargs['presence_penalty']
+        generate_kwargs[
+            'presence_penalty'] = None if presence_penalty == 0.0 else presence_penalty * torch.ones(
+                batch_size, dtype=torch.float32)
+        generate_kwargs['beam_search_diversity_rate'] *= torch.ones(
+            batch_size, dtype=torch.float32)
+        generate_kwargs['len_penalty'] *= torch.ones(size=[batch_size],
+                                                     dtype=torch.float32)
+        generate_kwargs['min_length'] *= torch.ones(size=[batch_size],
+                                                    dtype=torch.int32)
+        if generate_kwargs['random_seed']:
+            generate_kwargs['random_seed'] = torch.randint(0,
+                                                           10000,
+                                                           size=[batch_size],
+                                                           dtype=torch.int64)
+
+        return generate_input, generate_kwargs
+
+    def predict(self, **inputs: Dict[str, Any]):
+        generate_input, generate_kwargs = self._parse_inputs(inputs)
+        start_ids = [
+            torch.tensor(self.tokenizer.encode(c),
+                         dtype=torch.int32,
+                         device=self.device) for c in generate_input
+        ]
+        start_lengths = [len(ids) for ids in start_ids]
+        start_ids = pad_sequence(start_ids,
+                                 batch_first=True,
+                                 padding_value=self.end_id)
+        start_lengths = torch.IntTensor(start_lengths)
+        tokens_batch = self.model(start_ids, start_lengths, **generate_kwargs)
+        outputs = []
+        for tokens in tokens_batch:
+            for beam_id in range(generate_kwargs['beam_width']):
+                # Do not exclude context input from the output
+                # token = tokens[beam_id][start_lengths[i]:]
+                token = tokens[beam_id]
+                # stop at end_id; This is the same as eos_token_id
+                token = token[token != self.end_id]
+                output = self.tokenizer.decode(token)
+                outputs.append(output)
+        return outputs
+
+    def predict_stream(self, **inputs: Dict[str, Any]):
+        raise RuntimeError('Streaming is not supported with FasterTransformer!')
```

Comment:
Out of curiosity, is this not possible at all or just not something we're focusing on rn?